evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -535,3 +535,86 @@ def test_partial_dependence_all_nan_value_error(logistic_regression_binary_pipel
+    X = pd.DataFrame(X, columns=[str(i) for i in range(X.shape[1])])
+    y = pd.Series(y)
+    X['dt_column'] = pd.Series(pd.date_range('20200101', periods=X.shape[0]))
+
+    pipeline.fit(X, y)
+    part_dep = partial_dependence(pipeline, X, features='dt_column')
+    if problem_type == 'multiclass':
+        assert len(part_dep["partial_dependence"]) == 300  # 100 rows * 3 classes
+        assert len(part_dep["feature_values"]) == 300
+    else:
+        assert len(part_dep["partial_dependence"]) == 100
+        assert len(part_dep["feature_values"]) == 100
+    assert not part_dep.isnull().any(axis=None)
+
+    part_dep = partial_dependence(pipeline, X, features=20)
+    if problem_type == 'multiclass':
+        assert len(part_dep["partial_dependence"]) == 300  # 100 rows * 3 classes
+        assert len(part_dep["feature_values"]) == 300
+    else:
+        assert len(part_dep["partial_dependence"]) == 100
+        assert len(part_dep["feature_values"]) == 100
+    assert not part_dep.isnull().any(axis=None)
+
+    with pytest.raises(ValueError, match='Two-way partial dependence is not supported for datetime columns.'):
+        part_dep = partial_dependence(pipeline, X, features=('0', 'dt_column'))
+    with pytest.raises(ValueError, match='Two-way partial dependence is not supported for datetime columns.'):
+        part_dep = partial_dependence(pipeline, X, features=(0, 20))
+
+
+@pytest.mark.parametrize('problem_type', ['binary', 'regression'])
+def test_graph_partial_dependence_regression_and_binary_datetime(problem_type,
+                                                                 X_y_regression, X_y_binary,
+                                                                 logistic_regression_binary_pipeline_class):
+    pytest.importorskip('plotly.graph_objects', reason='Skipping plotting test because plotly not installed')
+
+    class LogisticRegressionBinaryPipeline(BinaryClassificationPipeline):
+        component_graph = ['Imputer', 'One Hot Encoder', 'DateTime Featurization Component', 'Standard Scaler', 'Logistic Regression Classifier']
+
+    class LinearRegressionPipeline(RegressionPipeline):
+        component_graph = ['Imputer', 'One Hot Encoder', 'DateTime Featurization Component', 'Standard Scaler', 'Linear Regressor']
+
+    if problem_type == 'binary':
+        X, y = X_y_binary
+        pipeline = LogisticRegressionBinaryPipeline({})
+    else:
+        X, y = X_y_regression
+        pipeline = LinearRegressionPipeline({})
+
+    X = pd.DataFrame(X, columns=[str(i) for i in range(X.shape[1])])
+    y = pd.Series(y)
+    X['dt_column'] = pd.to_datetime(pd.Series(pd.date_range('20200101', periods=X.shape[0])), errors='coerce')
+
+    pipeline.fit(X, y)
+
+    fig = graph_partial_dependence(pipeline, X, features='dt_column', grid_resolution=5)
+    plot_data = fig.to_dict()['data'][0]
+    assert plot_data['type'] == 'scatter'
+    assert plot_data['x'].tolist() == list(pd.date_range('20200101', periods=X.shape[0]))
```

Comment:
If the dates in the dataframe are not in sorted order, is the output in sorted order? I believe so, but if not, I think we may need to modify the implementation to make sure that's the case?

Reply:
I'll add a test to make sure!